components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,836 @@
+                    var validWeekRange = parseInt(dayOfWeekRangeArr[0]) < parseInt(dayOfWeekRangeArr[1]);
+                    var validWeekStrRange = weekArray.indexOf(dayOfWeekRangeArr[0]) < weekArray.indexOf(dayOfWeekRangeArr[1]);
+                    return !isNaN(parseInt(dayOfWeekRangeArr[0])) && !isNaN(parseInt(dayOfWeekRangeArr[1])) ?
+                                        validateRangeOfDay(dayOfWeekRangeArr, 1, 7) && validWeekRange :
+                                        validateStringValue(dayOfWeekRangeArr, weekArray) && validWeekStrRange;
+                } else if(dayOfWeek.includes('#') && dayOfMonth === '?') {
+                    var weekdayOfMonthArr = dayOfWeek.split('#');
+                    if(dayOfWeek.endsWith('#')){
+                        errorMsg = "Day-of-week values should not end with # "
+                    } else {
+                        errorMsg = "Day-of-Week values must be between 1 and 7";
+                    }
+                    return (validateStringValue([weekdayOfMonthArr[0]], weekArray) &&
+                                validateRangeOfDay([weekdayOfMonthArr[1]], 1, 5)) ||
+                                 (validateRangeOfDay([weekdayOfMonthArr[0]], 1, 7) &&
+                                 validateRangeOfDay([weekdayOfMonthArr[1]], 1, 5));
+                } else {
+                    if(dayOfWeek === ""){
+                        errorMsg = "Day-of-week values should not end with , "
+                    } else {
+                        errorMsg = "Day-of-Week values must be between 1 and 7";
+                    }
+                    var firstIndexValue = dayOfWeek.charAt(0);
+                    var secondIndexValue = dayOfWeek.charAt(1);
+                    if(dayOfWeek.length === 1){
+                        return validateRangeOfDay([dayOfWeek], 1, 7) || dayOfWeek === 'L' ||
+                            (dayOfWeek === '*' && dayOfMonth !== '*') || (dayOfWeek === '?' && dayOfMonth !== '?');
+                    } else {
+                        return (!isNaN(parseInt(firstIndexValue)) && !isNaN(parseInt(secondIndexValue))) ?
+                                validateRangeOfDay([dayOfWeek], 1, 7) : validateStringValue([dayOfWeek], weekArray);
+                    }
+                }
+            });
+        };
+        var validateCronExpression = function(cronExpression) {
+            if(!constants.REGEX_FOR_SPACE.test(cronExpression)) {
+                return false;
+            }
+            var cronArray = cronExpression.split(" ");
+            if(parseInt(cronArray.length) !== 6) {
+                return false;
+            }
+            var seconds = cronArray[0];
+            var minutes = cronArray[1];
+            var hours = cronArray[2];
+            var dayOfMonth = cronArray[3];
+            var month = cronArray[4];
+            var dayOfWeek = cronArray[5];
+
+            var isValidSeconds = validateTimeValue(seconds, constants.MAX_MINUTE_AND_SECOND_VALUE);
```

Comment:
here if any of them returns false you dont need to check others right . Use a if condition so it will not check all of them if any one of them is false

Reply:
fixed in e096cb9